frontend/src/metabase/lib/permissions.js

```diff
@@ -186,11 +186,12 @@ export function updateTablesPermission(permissions: GroupsPermissions, groupId:
 }
 
 export function updateSchemasPermission(permissions: GroupsPermissions, groupId: GroupId, { databaseId }: DatabaseEntityId, value: string, metadata: Metadata): GroupsPermissions {
-    let database = metadata.database(databaseId);
-    let schemaNames = database && database.schemaNames();
+    const database = metadata.database(databaseId);
+    const schemaNames = database && database.schemaNames();
+    const schemaNamesOrNoSchema = schemaNames.length > 0 ? schemaNames : [""];
```

Comment:
Nulls are filtered out in https://github.com/metabase/metabase/blob/afb6230312d44413de01124ace1b2b143362f802/frontend/src/metabase/meta/metadata/Database.js#L38